core/src/main/java/org/apache/carbondata/core/scan/expression/conditional/FilterUtil.java

```diff
@@ -0,0 +1,122 @@
+    } else if (DataTypes.INT.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.INT),
+          new LiteralExpression(value, DataTypes.INT));
+    } else if (DataTypes.DOUBLE.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.DOUBLE),
+          new LiteralExpression(value, DataTypes.DOUBLE));
+    } else if (DataTypes.FLOAT.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.FLOAT),
+          new LiteralExpression(value, DataTypes.FLOAT));
+    } else if (DataTypes.SHORT.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.SHORT),
+          new LiteralExpression(value, DataTypes.SHORT));
+    } else if (DataTypes.BINARY.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.BINARY),
+          new LiteralExpression(value, DataTypes.BINARY));
+    } else if (DataTypes.DATE.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.DATE),
+          new LiteralExpression(value, DataTypes.DATE));
+    } else if (DataTypes.LONG.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.LONG),
+          new LiteralExpression(value, DataTypes.LONG));
+    } else if (DataTypes.TIMESTAMP.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.TIMESTAMP),
+          new LiteralExpression(value, DataTypes.TIMESTAMP));
+    } else if (DataTypes.BYTE.getName().equalsIgnoreCase(dataType)) {
+      return new EqualToExpression(
+          new ColumnExpression(columnName, DataTypes.BYTE),
+          new LiteralExpression(value, DataTypes.BYTE));
+    } else {
+      throw new IllegalArgumentException("Unsupported data type: " + dataType);
+    }
+  }
+
+  public static Expression prepareEqualToExpressionSet(String columnName, DataType dataType,
+      List<Object> values) {
+    Expression expression = null;
+    if (0 == values.size()) {
+      expression = prepareEqualToExpression(columnName, dataType.getName(), null);
+    } else {
+      expression = prepareEqualToExpression(columnName, dataType.getName(), values.get(0));
+    }
+    for (int i = 1; i < values.size(); i++) {
+      Expression expression2 = prepareEqualToExpression(columnName,
+          dataType.getName(), values.get(i));
+      expression = new OrExpression(expression, expression2);
+    }
+    return expression;
+  }
+
+  public static Expression prepareEqualToExpressionSet(String columnName, String dataType,
+      List<Object> values) {
```

Comment:
I think each object can be different data type. your set handles only same data type columns, thats not good. make it generic

Reply:
prepareEqualToExpressionSet  for different data type is not good. we should provide OrExpression for different data type.
I add prepareOrExpression in new commit, we can prepare equal expression for each and use prepareOrExpression for different data type.

Reply:
If you introduce OrExpression, then no need of prepareEqualToExpressionSet. 
you can form list of equal to expressions and pass it to OrExpression.
please remove prepareEqualToExpressionSet   --> because it is still takes single data type.